src/test/java/com/iexec/worker/docker/DockerServiceTests.java

```diff
@@ -283,9 +286,156 @@ public void shouldRunAndNotRemoveContainerFromRecordsSinceInDetachedMode() {
         verify(dockerService, never()).removeFromRunningContainersRecord(containerName);
     }
 
-    /**
-     * addToRunningContainersRecord()
-     */
+    //#region getInputBind()
+
+    @Test
+    public void shouldGetInputBind() {
+        String taskInputDir = "/input/dir";
+        when(workerConfigService.getTaskInputDir(CHAIN_TASK_ID)).thenReturn(taskInputDir);
+        assertThat(dockerService.getInputBind(CHAIN_TASK_ID))
+                // "/input/dir:/iexec_in"
+                .isEqualTo(taskInputDir + ":" + File.separator + "iexec_in");
+    }
+
+    //#endregion
+
+    //#region getIexecOutBind()
+
+    @Test
+    public void shouldGetIexecOutBind() {
+        String taskIexecOutDir = "/iexec/out/dir";
+        when(workerConfigService.getTaskIexecOutDir(CHAIN_TASK_ID)).thenReturn(taskIexecOutDir);
+        assertThat(dockerService.getIexecOutBind(CHAIN_TASK_ID))
+                // "/iexec/out/dir:/iexec_out"
+                .isEqualTo(taskIexecOutDir + ":" + File.separator + "iexec_out");
+    }
+
+    //#endregion
+
+    //#region stopAllRunningContainers
+
+    @Test
+    public void shouldStopAllRunningContainers() {
+        String container1 = "container1";
+        String container2 = "container2";
+        dockerService.addToRunningContainersRecord(container1);
+        dockerService.addToRunningContainersRecord(container2);
+
+        dockerService.stopAllRunningContainers();
+        // Verify all containers are removed
+        verify(dockerService, times(2)).stopRunningContainer(anyString());
+        verify(dockerService).stopRunningContainer(container1);
+        verify(dockerService).stopRunningContainer(container2);
+    }
+
+    @Test
+    public void shouldNotStopRunningContainers() {
+        // no running container
+        doReturn(dockerClientInstanceMock).when(dockerService).getClient();
+        dockerService.stopAllRunningContainers();
+        verify(dockerClientInstanceMock, never()).stopContainer(anyString());
+    }
+
+    //#endregion
+
+    //#region stopTaskRunningContainers()
+
+    @Test
+    public void shouldStopTaskRunningContainers() {
+        String computeContainerName = "awesome-app-" + CHAIN_TASK_ID;
+        String preComputeContainerName = computeContainerName + "-tee-pre-compute";
+        String postComputeContainerName = computeContainerName + "-tee-post-compute";
+        // Add task related containers to record
+        dockerService.addToRunningContainersRecord(preComputeContainerName);
+        dockerService.addToRunningContainersRecord(computeContainerName);
+        dockerService.addToRunningContainersRecord(postComputeContainerName);
+        // Add some other tasks containers
+        dockerService.addToRunningContainersRecord("containerName1");
+        dockerService.addToRunningContainersRecord("containerName2");
+
+        dockerService.stopTaskRunningContainers(CHAIN_TASK_ID);
+        // verify we removed all containers
```

Comment:
"verify we didn't remove more than task-related containers" ? We don't remove all containers, there are still 2 others running.

Reply:
Updated.